sources/api/apiserver/src/server/controller.rs

```diff
@@ -420,21 +498,91 @@ pub(crate) fn get_metadata_for_data_keys<D: DataStore, S: AsRef<str>>(
     Ok(result)
 }
 
-/// Gets the value of a metadata key everywhere it's found in the data store.  Returns a mapping
-/// of data key to the metadata value associated with the requested key.
-pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+// Sort the given Hashmap in topological order
+fn get_sorted_metadata(
+    metadata: HashMap<Key, HashMap<Key, String>>,
+) -> Vec<(Key, HashMap<Key, String>)> {
+    let mut metadata_sorted: Vec<_> = metadata.into_iter().collect();
+    metadata_sorted.sort_by(|(k1, _), (k2, _)| {
+        k1.segments()
+            .len()
+            .cmp(&k2.segments().len())
+            .then_with(|| k1.name().cmp(k2.name()))
+    });
+    metadata_sorted.to_vec()
+}
```

Comment:
I get why you are sorting by number of segments first.

Would it affect correctness to drop the secondary sort based on name? It seems like your intent is to establish a canonical ordering, with stronger properties than a topological ordering. If so then maybe update the comment accordingly.